### PR TITLE
Pack EPID group inside Enclave structure and delete exposed FFI layer…

### PR DESCRIPTION
**Pull request by manju956**
… API to get epid group

Signed-off-by: manju.ac <manjunath.a.c@intel.com>

### CODE REVIEW

File: src/sgx/libpoet_bridge/enclave.cpp [hunk 1]
```diff
@@ -18,6 +18,8 @@
 #include <iostream>
 #include <sstream>
 #include <stdexcept>
+#include <iomanip>
```
Comment (arsulegai) [inline src/sgx/libpoet_bridge/enclave.cpp]: Are these include statements being used?

File: src/sgx/libpoet_bridge/enclave.cpp [hunk 2]
```diff
@@ -760,5 +744,5 @@ namespace sawtooth {
             }
         } // Enclave::QuerySgxStatus
 
-    } // namespace poet
+     }// namespace poet
```
Comment (arsulegai) [inline src/sgx/libpoet_bridge/enclave.cpp]: [nitpick] fmt issue

File: src/sgx/libpoet_bridge/enclave.cpp [hunk 3]
```diff
@@ -108,51 +111,27 @@ namespace sawtooth {
...
+            sgx_epid_group_id_t* outEpidGroup
             )
         {
             ThrowIfNull(
                 outEnclaveMeasurement,
                 "Enclave measurement pointer is NULL");
+            ThrowIfNull(
+                outEpidGroup,
+                "Enclave EPID group pointer is NULL");
+
```
Comment (arsulegai) [inline src/sgx/libpoet_bridge/enclave.cpp]: [nitpick] fmt issue

File: src/sgx/libpoet_bridge/poet.cpp [hunk 4]
```diff
@@ -291,6 +255,18 @@ poet_err_t Poet_GetEnclaveCharacteristics(
            inEnclaveBasenameLength,
            hexString.c_str(),
            hexString.length());
+
+        std::reverse(epidGroup, epidGroup + sizeof(epidGroup));
```
Comment (arsulegai) [inline src/sgx/libpoet_bridge/poet.cpp:259]: [Question]: epidGroup is of type sgx_epid_group_id_t  and sgx_epid_group_id_t  is array of uint8_t with size 4. sizeof(epidGroup) would then return 1 byte * 4 length which is 4. So intent here is to reverse order elements in array epidGroup. Could you please correct me if I didn't understand something here?

File: src/sgx/rust_sgxffi/src/lib.rs [hunk 5]
```diff
@@ -74,13 +77,6 @@ mod tests {
             ffi::init_enclave(&mut eid, bin_path, spid_str).expect("Failed to initialize enclave");
         assert_eq!(ret, "Success");
 
-        let mut epid_info: r_sgx_epid_group_t = r_sgx_epid_group_t {
```
Comment (arsulegai) [inline src/sgx/rust_sgxffi/src/lib.rs:77]: Can we somehow assert here and make sure epidGroupId is always read?

File: src/sgx/libpoet_bridge/enclave.cpp (hunk 1 above)
Comment (manju956) [reply src/sgx/libpoet_bridge/enclave.cpp]: these include are not used, will remove them

File: src/sgx/libpoet_bridge/poet.cpp (hunk 4 above)
Comment (manju956) [reply src/sgx/libpoet_bridge/poet.cpp:259]: Yes, the understanding is correct

**Review by arsulegai — APPROVED** [review]

File: src/sgx/libpoet_bridge/poet.cpp (hunk 4 above)
Comment (arsulegai) [reply src/sgx/libpoet_bridge/poet.cpp:259]: Could you please add a comment here along with unit test correction patch? (something like changing the epidGroup to big endian)

**Pull request merged**